forest/src/cli/chain_cmd.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::Cid;
+use jsonrpc_v2::Error as JsonRpcError;
+use log::warn;
+use rpc_client::{block, genesis, head, messages, new_client, read_obj};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum ChainCommands {
+    /// Retrieves and prints out the block specified by the given CID
+    #[structopt(about = "<Cid> Retrieve a block and print its details")]
+    Block {
+        #[structopt(help = "Input a valid CID")]
+        cid: String,
+    },
+
+    /// Prints out the genesis tipset
+    #[structopt(about = "Prints genesis tipset", help = "Prints genesis tipset")]
+    Genesis,
+
+    /// Prints out the canonical head of the chain
+    #[structopt(about = "Print chain head", help = "Print chain head")]
+    Head,
+
+    /// Reads and prints out a message referenced by the specified CID from the
+    /// chain blockstore
+    #[structopt(about = "<CID> Retrieves and prints messages by CIDs")]
+    Message {
+        #[structopt(help = "Input a valid CID")]
+        cid: String,
+    },
+
+    /// Reads and prints out ipld nodes referenced by the specified CID from chain
+    /// blockstore and returns raw bytes
+    #[structopt(about = "<CID> Read the raw bytes of an object")]
+    ReadObj {
+        #[structopt(help = "Input a valid CID")]
+        cid: String,
+    },
+}
+
+impl ChainCommands {
+    pub async fn run(&self) {
+        // TODO handle cli config
+        match self {
+            Self::Block { cid } => {
+                let cid: Cid = cid.parse().unwrap();
+                let client = new_client();
+
+                let blk = block(client, cid)
+                    .await
+                    .map_err(|e| {
+                        stringify_rpc_err(e);
+                    })
+                    .unwrap();
```

Comment:
For the sync commands I'm handling the unwraps and outputting messages when certain commands fail. I get that this is just a CLI, and it doesn't persist, so error handling isnt a big issue. Just wanted other peoples thoughts, should we deal with the unwraps ?

Reply:
the benefit is a cleaner message output, but at the cost of having to handle the error and exit gracefully. do it if it's easy to setup, but it isn't necessary for this sort of thing